astroid/brain/brain_builtin_inference.py

```diff
@@ -515,6 +516,62 @@ def _infer_object__new__decorator_check(node):
     return False
 
 
+def infer_isinstance(callnode, context=None):
+    """Infer isinstance calls
+
+    :param nodes.Call callnode: an isinstance call
+    :rtype nodes.Const: Boolean Const value of isinstance call
+    """
+    call = arguments.CallSite.from_call(callnode)
+    if call.keyword_arguments:
+        # isinstance doesn't support keyword arguments
+        raise TypeInferenceError("isinstance() takes no keyword arguments")
```

Comment:
Since this is part of the inference, we raise `UseDefaultInference` instead of an error. By raising an error, every time we infer `isinstance` as part of, say pylint, we're going to crash rather than throwing an `InferenceError` or returning `Uninferable`. For the cases where something is invalid in the inference engine, like this one, we usually add pylint rules.

Reply:
Alright. I don't like astroid silently failing (it makes debugging harder), but I'm not sure what to do about it at the moment.

Reply:
Yeah, I agree with the sentiment, but since pylint is the main consumer of astroid, it's not easy to make these changes.

As an example of a place where I wanted to get type error information out of astroid is in the binary operations inference (https://github.com/PyCQA/astroid/blob/3fae32f99c7f432a1281b151ea17e163a2e8dc47/astroid/inference.py#L659), where the type errors hints can later on be accessed by a `type_errors` method (https://github.com/PyCQA/astroid/blob/8425d915c0186f87da66ad6223588ce70c49031c/astroid/node_classes.py#L1779). It's super ugly, by pylint can acess this one without having the entire inference crash down on a type error somewhere in a random file.

Reply:
I guess I would have to do this in `Call`. I might come back to this later